core/scripts/Voting.js

```diff
@@ -114,14 +142,16 @@ class VotingSystem {
       identifier: request.identifier,
       time: request.time,
       hash,
-      encryptedVote
+      encryptedVote,
+      price: fetchedPrice,
```

Comment:
I'm surprised that `batchCommit` works with these additional fields. Does web3 drop these fields when generating the transaction? I guess my only concern is that `price` and `salt` are part of the transaction.

Reply:
As discussed IRL, web3 does drop them - it errors for missing fields, but not extra ones. I also double checked the transaction data, and it is identical whether extra fields are added vs not. However, I think it makes sense to be explicit and filter these out before passing along to web3 just to be safe for future versions.